### PR TITLE
V2 documents now use boto s3 download, not raw http requests

### PR DESCRIPTION
**Pull request by dragon-dxw**

### CODE REVIEW

File: ds-caselaw-ingester/lambda_function.py [hunk 1]
```diff
@@ -246,26 +273,26 @@ def copy_file(tarfile, input_filename, output_filename, uri, s3_client: Session.
...
-            sqs_client.send_message(
-                QueueUrl=os.getenv("SQS_QUEUE_URL"),
-                MessageBody=json.dumps(retry_message),
-            )
-        else:
-            raise MaximumRetriesExceededException(
-                f'Maximum number of retries reached for {original_message["consignment-reference"]}'
-            )
-    else:
+    if not is_v1(original_message):
```
Comment (dragon-dxw) [inline ds-caselaw-ingester/lambda_function.py:276]: All we do here is flip the control to reduce nesting.

File: ds-caselaw-ingester/lambda_function.py [hunk 2]
```diff
@@ -367,22 +394,7 @@ def handler(event, context):
...
-        # Send retry message to sqs if the GET fails
-        send_retry_message(message, sqs_client)
-        raise
-    # Store it in the /tmp directory
-    filename = os.path.join("/tmp", f"{consignment_reference}.tar.gz")
-
-    with open(filename, "wb") as out:
-        out.write(tar_gz_contents)
-        out.close()
+    filename = save_s3_response(message, sqs_client, s3_client)
```
Comment (dragon-dxw) [inline ds-caselaw-ingester/lambda_function.py:397]: the deleted code here is moved into the function.

File: ds-caselaw-ingester/tests.py [hunk 3]
```diff
@@ -521,8 +521,8 @@ def test_get_consignment_reference_empty_v2(self):
         message["parameters"][
             "bundleFileURI"
         ] = "http://172.17.0.2:4566/te-editorial-out-int/ewca_civ_2021_1881.tar.gz"
-        result = lambda_function.get_consignment_reference(message)
-        assert result == "ewca_civ_2021_1881"
+        with pytest.raises(lambda_function.InvalidMessageException):
```
Comment (dragon-dxw) [inline ds-caselaw-ingester/tests.py:524]: Parsing the consignment reference from the filename was always a hack; we have changed the assumption to "the consignment reference always exists"

File: aws_examples/sns/parsed-judgment-v2.json [hunk 4]
```diff
@@ -1,18 +1,19 @@
 {
   "properties": {
-      "messageType": "uk.gov.nationalarchives.tre.messages.judgmentpackage.available.JudgmentPackageAvailable",
```
Comment (dragon-dxw) [inline aws_examples/sns/parsed-judgment-v2.json:3]: we replace the old v2 judgment with a more accurate version

File: scripts/test [hunk 5]
```diff
@@ -1 +1 @@
-MARKLOGIC_HOST= MARKLOGIC_USER= MARKLOGIC_PASSWORD= python -m pytest ds-caselaw-ingester/tests.py
+MARKLOGIC_HOST= MARKLOGIC_USER= MARKLOGIC_PASSWORD= python -m pytest ds-caselaw-ingester/tests.py $*
```
Comment (dragon-dxw) [inline scripts/test:1]: lets us put `-v` at the end of scripts/test to get verbose output.

File: ds-caselaw-ingester/tests.py [hunk 6]
```diff
@@ -94,6 +119,7 @@ def test_handler_messages_v2(
         log = capsys.readouterr().out
         assert "Ingester Start: Consignment reference FCL-12345" in log
         assert "v1: False" in log
+        assert "tar.gz saved locally as /tmp/FCL-12345.tar.gz" in log
```
Comment (anthonyhashemi) [inline ds-caselaw-ingester/tests.py:122]: nice

File: ds-caselaw-ingester/tests.py [hunk 7]
```diff
@@ -41,6 +48,10 @@
 v2_message = json.loads(v2_message_raw)
 
 
+def create_fake_file(*args, **kwargs):
```
Comment (anthonyhashemi) [inline ds-caselaw-ingester/tests.py:51]:
```suggestion
def create_fake_TDR_tarball(*args, **kwargs):
```

File: ds-caselaw-ingester/tests.py [hunk 8]
```diff
@@ -16,6 +17,12 @@
 )
 from notifications_python_client.notifications import NotificationsAPIClient
 
+TDR_TARBALL_PATH = os.path.join(
```
[the remark this answers is not in the document]
Comment (anthonyhashemi) [reply ds-caselaw-ingester/tests.py:20]: This is outside the class. I guess you could just delete the other one within the class and point to this

File: scripts/setup-localstack.sh [hunk 9]
```diff
@@ -36,4 +36,7 @@ else
   awslocal s3 cp aws_examples/s3/te-editorial-out-int/TDR-2022-DNWR.tar.gz s3://te-editorial-out-int
 fi
 
+awslocal s3api create-bucket --bucket staging-tre-court-document-pack-out
+awslocal s3 cp aws_examples/s3/te-editorial-out-int/TDR-2022-DNWR.tar.gz s3://staging-tre-court-document-pack-out/QX/e31b117f-ff09-49b6-a697-7952c7a67384/QX.tar.gz
```
[the remark this answers is not in the document]
Comment (dragon-dxw) [reply scripts/setup-localstack.sh:40]: Yep.

File: ds-caselaw-ingester/tests.py (hunk 7 above)
Comment (dragon-dxw) [reply ds-caselaw-ingester/tests.py]: good call.

**Review by anthonyhashemi — APPROVED** [review]
Have left a couple of suggestions, but generally looks good 👍

**Pull request merged**